src/main/cpp/include/utils/vid_mapper_sql.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * The MIT License (MIT)
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+#ifdef LIBDBI
+#ifndef GENOMICSDB_VID_MAPPER_SQL_H
+#define GENOMICSDB_VID_MAPPER_SQL_H
+
+#include "vid_mapper.h"
+#include <string>
+#include <vector>
+#include <map>
+#include <dbi/dbi.h>
+
+enum {
+  GENOMICSDB_VID_MAPPER_SUCCESS = 0x0,
+  GENOMICSDB_VID_MAPPER_FAILURE = 0x1
+};
+
+class SQLVidMapperRequest {
+  public:
+    std::string host_name;
+    std::string user_name;
+    std::string pass_word;
+    std::string db_name;
+    std::string work_space;
+    std::string array_name;
+};
+
+class SQLBasedVidMapper : public VidMapper {
+  public:
+    SQLBasedVidMapper(const SQLVidMapperRequest&);
+
```

Comment:
We should probably delete the copy constructor since there is no way to do deep copy of the libdbi structures.
If you wish to create a move constructor, you can probably copy the libdbi fields to the current object and zero out the other structure.
To delete the copy and move constructors, do:
SQLBasedVidMapper(const SQLBasedVidMapper& other) = delete;
SQLBasedVidMapper(SQLBasedVidMapper&& other) = delete;